op-deployer/pkg/deployer/state/chain_intent.go

```
@@ -119,9 +127,32 @@ func (c *ChainIntent) Check() error {
 		return fmt.Errorf("%w: chainId=%s", ErrFeeVaultZeroAddress, c.ID)
 	}
 
+	if c.CustomGasToken.Enabled {
+		if c.CustomGasToken.Name == "" {
+			return fmt.Errorf("%w: CustomGasToken.Name cannot be empty when enabled, chainId=%s", ErrIncompatibleValue, c.ID)
+		}
+		if c.CustomGasToken.Symbol == "" {
+			return fmt.Errorf("%w: CustomGasToken.Symbol cannot be empty when enabled, chainId=%s", ErrIncompatibleValue, c.ID)
+		}
+
+		if c.CustomGasToken.NativeAssetLiquidityAmount == nil {
+			return fmt.Errorf("%w: CustomGasToken.NativeAssetLiquidityAmount must be set when custom gas token is enabled, chainId=%s", ErrIncompatibleValue, c.ID)
+		}
```

Should we also ensure this is a positive big.Int?

---

We decided to not make it a strict requirement since maybe there can be an use case where the chain wants to start with zero. Either way the liquidity can be funded later directly to the `NativeAssetLiquidity` contract through an upgrade tx

---

Ok sounds like 0 value is acceptable. Should we ensure its not negative?

---

We are assuming that's not needed since the L2 genesis script will revert in that case

---

This chainIntent is created before the L2 genesis script runs though. I think it would be a nice protection to add. Better to fail early than far downstream. This is the op-deployer flow:

1. `op-deployer init` - creates intent.toml
2. user populates intent.toml (which contains the chainIntent structs)
3. `op-deployer apply` - executes the following sequential pipeline
    1. `DeploySuperchain.s.sol` (if needed)
    2. `DeployImplementations.s.sol` (if needed) 
    3. `DeployOPChain.s.sol` 
    4. `L2Genesis.s.sol`

---

Fixed https://github.com/ethereum-optimism/optimism/pull/17412/commits/4ad8a301948a09f77c2d239ef15604d76fd8546c